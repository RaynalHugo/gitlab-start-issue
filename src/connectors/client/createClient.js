import axios from "axios";
import { isNil } from "lodash/fp";

export function createClient({ userPrivateToken, gitlabUrl }) {
	if (!isNil(userPrivateToken)) {
		return axios.create({
			baseURL: gitlabUrl,
			timeout: 10000,
			headers: { "Private-Token": userPrivateToken }
		});
	} else {
		throw new Error(
			"userPrivateToken is not definied in client `createClient`"
		);
	}
}

export const client = createClient({
	gitlabUrl: "https://gitlab.com/api/v4",
	userPrivateToken: "ddx1wWi8xo6ryc4XbBqj"
});
