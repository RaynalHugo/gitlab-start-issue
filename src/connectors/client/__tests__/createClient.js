import { createClient } from "../createClient";

test("createClient", () => {
	const client = createClient({
		gitlabUrl: "https://gitlab.com/api/v4",
		userPrivateToken: "ddx1wWi8xo6ryc4XbBqj"
	});
	expect(client).toMatchSnapshot();
});
