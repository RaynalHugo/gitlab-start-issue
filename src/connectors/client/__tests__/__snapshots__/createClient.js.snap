// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createClient 1`] = `[Function]`;
